src/test/java/com/mathworks/ci/MatlabInstallationTest.java

```diff
@@ -21,10 +28,40 @@
 public class MatlabInstallationTest {
 
     private MatlabInstallation.DescriptorImpl matlabInstDescriptor;
+    private static URL url;
+    private static String FileSeperator;
+    private static String VERSION_INFO_XML_FILE = "VersionInfo.xml";
+    private static String matlabExecutorAbsolutePath;
 
     @Rule
     public JenkinsRule jenkins = new JenkinsRule();
 
+    @BeforeClass
+    public static void classSetup() throws URISyntaxException, IOException {
+        ClassLoader classLoader = RunMatlabTestsBuilderTest.class.getClassLoader();
+        if (!System.getProperty("os.name").startsWith("Win")) {
+            FileSeperator = "/";
+            url = classLoader.getResource("com/mathworks/ci/linux/bin/matlab.sh");
+            try {
+                matlabExecutorAbsolutePath = new File(url.toURI()).getAbsolutePath();
+
+                // Need to do this operation due to bug in maven Resource copy plugin [
+                // https://issues.apache.org/jira/browse/MRESOURCES-132 ]
+
+                ProcessBuilder pb = new ProcessBuilder("chmod", "755", matlabExecutorAbsolutePath);
+                pb.start();
+            } catch (IOException e) {
+                e.printStackTrace();
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
+            }
+        } else {
+            FileSeperator = "\\";
+            url = classLoader.getResource("com/mathworks/ci/win/bin/matlab.bat");
+            matlabExecutorAbsolutePath = new File(url.toURI()).getAbsolutePath();
+        }
+    }
+
```

Comment:
Not sure why line 65 onwards is not showing as newly added block. I hope its newly added class all together

Reply:
It is a pre-existing class, I added newer tests to it.